source/app/libpd.c

```diff
@@ -216,6 +251,111 @@ void parodus_receive_wait()
         WalPrint ("End of parodus_upstream\n");
 }
 
+char *get_global_cloud_status()
+{
+    return cloud_status;
+}
+
+void set_global_cloud_status(char *status)
+{
+    cloud_status = status;
+}
+
+//send cloud-status upstream RETRIEVE request to parodus to check connectivity
+int getConnCloudStatus(char *device_mac)
+{
+	char *source = NULL, *dest = NULL;
+	wrp_msg_t *req_wrp_msg = NULL;
+	char *contentType = NULL;
+	int rv = -1;
+	int sendStatus = -1;
+    int backoffRetryTime = 0;
+    int c=2;
+
+	if(device_mac == NULL)
+	{
+		WalError("device_mac is NULL, unable to get cloud_status\n");
+		return rv;
+	}
+	else
+	{
+		req_wrp_msg = (wrp_msg_t *)malloc(sizeof(wrp_msg_t));
+		if(req_wrp_msg != NULL)
+		{
+			memset(req_wrp_msg, 0, sizeof(wrp_msg_t));
+			req_wrp_msg->msg_type = WRP_MSG_TYPE__RETREIVE;
+
+			source = (char *) malloc(sizeof(char)*MAX_STR_LENGTH);
+			dest   = (char *) malloc(sizeof(char)*MAX_STR_LENGTH);
+
+			if(source !=NULL)
+			{
+				snprintf(source, MAX_STR_LENGTH, "mac:%s/config", device_mac);
+				req_wrp_msg->u.crud.source = source;
+				WalPrint("req_wrp_msg->u.crud.source is %s\n", req_wrp_msg->u.crud.source);
+			}
+
+			if(dest !=NULL)
+			{
+				snprintf(dest, MAX_STR_LENGTH, "mac:%s/parodus/cloud-status", device_mac);
+				req_wrp_msg->u.crud.dest = dest;
+				WalPrint("req_wrp_msg->u.crud.dest is %s\n", req_wrp_msg->u.crud.dest);
+			}
+
+			req_wrp_msg->u.crud.transaction_uuid = strdup("bd4ad2d1-5c9c-486f-8e25-52c242b38");
```

Comment:
Why is this transaction id hardcoded?

Reply:
For CRUD requests, user should pass transaction_id as server is not generating its own and it is returning empty payload response. 

To handle this, added a method to generate random transaction_uuid for every cloud-status retrieve request.